tools/nl/translate_compare/translate_csvs.py

```diff
@@ -0,0 +1,220 @@
+    DEFAULT_NEW_COLUMN_NAME,
+    help="header of new column with outputted translations",
+)
+flags.DEFINE_integer(
+    "source_column_index",
+    None,
+    help="index of column to translate (zero-indexed)",
+)
+flags.DEFINE_string(
+    "source_column_name",
+    None,
+    help="header of column to translate",
+)
+flags.DEFINE_string(
+    "output_language",
+    DEFAULT_OUTPUT_LANUGAGE_CODE,
+    help="2 letter code of the language to translate to",
+)
+flags.DEFINE_string(
+    "input_language",
+    None,
+    help="2 letter code of the language to translate from"
+)
+
+
+class Translator:
+    def __init__(self):
+        self.client = translate.TranslationServiceClient()
+
+        # get list of supported languages
+        languages_response = self.client.get_supported_languages(
+            parent=PARENT,
+            display_language_code="en",
+        )
+        self.supported_languages = [
+            lang.language_code for lang in languages_response.languages
+        ]
+
+    def translate_single_query(
+        self, text: str, input_lang: str = None, output_lang: str = "en"
+    ) -> str:
+        """Translate a single query using the Google Translate API.
+
+        Sourced from: https://codelabs.developers.google.com/codelabs/cloud-translation-python3#4
+
+        Args:
+            text: query to translate
+            input_lang: source language, as 2 letter language code
+                        if not provided, will use translation API's detect language feature
+            output_lang: target language, as 2 letter language code
+                        defaults to English
+
+        Returns:
+            translated text
+        """
+        response = self.client.translate_text(
+            parent=PARENT,
+            contents=[text],
+            source_language_code=input_lang,
+            target_language_code=output_lang,
+        )
+        return response.translations[0].translated_text
+
+    def translate_multiple_queries(
+        self,
+        texts: list[str],
+        input_lang: str = None,
+        output_lang: str = "en",
+    ) -> list[str]:
+        """Translate multiple queries using the Google Translate API.
+
+        Sourced from: https://codelabs.developers.google.com/codelabs/cloud-translation-python3#4
+
+        Args:
+            texts: queries to translate
+            input_lang: source language, as 2 letter language code
+                        if not provided, will use translation API's detect language feature
+            output_lang: target language, as 2 letter language code
+                        defaults to English
+
+        Returns:
+            translated text
+        """
+        response = self.client.translate_text(
+            parent=PARENT,
+            contents=texts,
+            source_language_code=input_lang,
+            target_language_code=output_lang,
+        )
+        translations = [resp.translated_text for resp in response.translations]
```

Comment:
Couple questions regarding the translation API: Are the translations guaranteed to be the same size and in the same order as the input texts

Reply:
Looks like it does: https://screenshot.googleplex.com/3ehF92MS6Rf3Bjk